regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
-    )
+    if (input$line_type == 'routes') {
+      leafletProxy("map") %>% addLayersControl(
+        position = c("bottomright"),
+        overlayGroups = c("routes_fast", "routes_quieter"),
+        options = layersControlOptions(collapsed = T)
+      )
+    } else if (input$line_type != 'routes') {
+      leafletProxy("map") %>% showGroup(c("routes_fast", "routes_quieter")) %>% removeLayersControl()
+    }
+  })
 
-    output$download_z_csv <- downloadHandler(
-      filename = function() { "zones.csv"  },
-      content = function(file) { write.csv(signif_sdf(to_plot$zones)@data[codebook_z$`Variable name`], file = file) }
-    )
+  ## Read region_stats.html, if it exists, for the loaded region
+  output$region_stats <- renderUI({
+    input$purpose
+    input$geography
+    region$current
 
-    output$download_z_geojson <- downloadHandler(
-      filename = function() { "zones.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$zones[codebook_z$`Variable name`]), file = file) }
-    )
+    region_stats_file <-
+      file.path("../tabs/region_stats", input$purpose, input$geography, region$current,
+                "region_stats.html")
+    if (file.exists(region_stats_file))
+      includeHTML(region_stats_file)
+    else
+      HTML("<strong>No statistics available</strong>")
+  })
 
-    output$download_l_geojson <- downloadHandler(
-      filename = function() { "lines.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$straight_line[codebook_l$`Variable name`]), file = file) }
-    )
+  ## Read regional data download files
+  # This file updates whenever there is a change to input$region, and content of links also depends on repo_sha
+  output$download_region_current <- renderUI({
+    region$current
 
-    output$download_rf_geojson <- downloadHandler(
-      filename = function() { "routes_fast.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$faster_route[codebook_r$`Variable name`]), file = file) }
-    )
+    # Create region name for files to be downloaded, so that they are accessible from the download_region.Rmd file
+    download_region <- data.frame(region_name = region$current)
+    saveRDS(download_region, file="../tabs/download_params_region_current.rds")
 
-    output$download_rq_geojson <- downloadHandler(
-      filename = function() { "routes_quiet.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$quieter_route[codebook_r$`Variable name`]), file = file) }
+    knitr::knit2html(quiet = T,
+                     input = file.path("../tabs/download_region.Rmd"),
+                     output = file.path("../tabs/download_region_current.html"),
+                     envir = globalenv(), force_v1 = T
     )
+    # Re-read html to remove style
+    download_region_file <- file.path("../tabs/download_region_current.html")
+    download_region <- readLines(download_regio
```

Comment:
I'd love to remove the data tables, I spent a long time getting them to work and would be happy if we didn't have to maintain them, has this been agreed on then?

Reply:
They could go if difficult to maintain but I think they are advantageous. Imagine if I don't have any computing skills but want to find the areas with the most car drivers. I could do this with a data table:

![image](https://user-images.githubusercontent.com/1825120/30031521-bf5d82f2-9189-11e7-972c-f9d7cf0fdfd7.png)

Worth checking with others. @AnnaGoodman1 @usr110 @JDWoodcock  what are your views?

Reply:
I think they should go, nice though they are in some ways.  The issue is that the data download pages  have now been considerably improved/extended in other ways, including now combining the lines and zones tabs to a single 'region data' page, and on that page showing all  available data downloads not just what is currently showing on the screen. 

I did think about it, and couldn't see anywhere to put these tables into the new data download pages without it being cluttered and confusing: esp since we would need both the zones and lines tables there on the same page.  The only good option I can see would be to have a show/hide-on-click function underneath the bullet points where we have e.g. region zone download data, saying 
"Click here to view an on-screen data table for selected information for the zones currently displayed on the map". then ditto for zones. but that sounds potentially difficult to build/maintain, plus having to create it for e.g. schools in due course. So my vote would be to remove this.

( Also even without many computing skills, downloading the relevant CSV 'all zones/ all flows' file  and doing a sort in that by the 'no cars' column should not be too hard?)

Reply:
Sounds reasonable, cheers for the input: makes good sense to me. Another issue with them is that the zones do not have names that are meaningful to most people and cannot be displayed on the map this way so the advantage is pretty minimal.